features/net/network-socket/WiFiInterface.h

```diff
@@ -17,22 +17,24 @@
 #ifndef WIFI_INTERFACE_H
 #define WIFI_INTERFACE_H
 
+#include <string.h>
 #include "network-socket/NetworkInterface.h"
 
+typedef struct wifi_ap {
+    char ssid[33]; /* 32 is what 802.11 defines as longest possible name; +1 for the \0 */
+    uint8_t bssid[6];
+    nsapi_security_t security;
+    int16_t rssi;
+    uint8_t channel;
 
-/** Enum of WiFi encryption types
- *
- *  The security type specifies a particular security to use when
- *  connected to a WiFi network
- *
- *  @enum nsapi_protocol_t
- */
-enum nsapi_security_t {
-    NSAPI_SECURITY_NONE = 0,   /*!< open access point */
-    NSAPI_SECURITY_WEP,        /*!< phrase conforms to WEP */
-    NSAPI_SECURITY_WPA,        /*!< phrase conforms to WPA */
-    NSAPI_SECURITY_WPA2,       /*!< phrase conforms to WPA2 */
-};
+    wifi_ap()
```

Comment:
Why a zeroing constructor over relying on [value-initialization](http://en.cppreference.com/w/cpp/language/value_initialization)?

Anywhere a zeroed `wifi_ap_t` is needed, the following should create a zeroed `wifi_ap_t`:
``` cpp
wifi_ap_t ap = wifi_ap_t();
```